### ISSUE TITLE
Automatic entry generation

### ISSUE THREAD

**Issue by JaydenBea** [annazwicker/cat-partners#54]
The website must be able to automatically generate entries. Ideally, they should be generated for a certain time period past the current date; say, two weeks. I'm going to add this functionality to the feeder page, so that when it's loaded, it ensures that such entries exist. Ideally this would be handled by a server between the client and the database, but we don't have that at present.

**Comment by JaydenBea** [comment]
2:30pm (start)

The first thing to do will be to move certain methods from `FirebaseHelper` to `Snapshots`, to keep in line with the process I started in #48. Some methods already exist that can help with this issue, but none of them are called yet.

**Comment by JaydenBea** [comment]
3pm (since 2:30pm)

Moved the required methods to `Snapshots`, and modified some of them to fix things. Working on a method called `ensureEntriesPast({int numDays = 14})` which will ensure entries exist up to `numDays` past the current date.

**Comment by JaydenBea** [comment]
3:25pm (since 3pm)

Finished the `ensureEntriesPast` method. It ensures entries exist for up to 14 days (by default) past the current date. It finds the latest date of entries in the database before performing generation, to ensure it doesn't "skip" dates. The method is invoked upon viewing the feeder page. I triggered it by viewing the page on my end; the page took longer than usual to load, but displayed successfully after the entries were generated. 

I also changed the column width of the 'date' column to use an `IntrinsicColumnWidth`; it had a static width before, which made the May dates display on two lines.

![image](https://github.com/annazwicker/cat-partners/assets/110794270/21104f5b-ab9d-4162-baad-77d14730384a)
